packages/wrangler/src/__tests__/configuration.test.ts

```diff
@@ -728,12 +728,69 @@ describe("normalizeAndValidateConfig()", () => {
           - Expected \\"jsx_factory\\" to be of type string but got 999.
           - Expected \\"jsx_fragment\\" to be of type string but got 1000.
           - Expected \\"tsconfig\\" to be of type string but got true.
-          - Expected \\"name\\" to be of type string but got 111.
+          - Expected \\"name\\" to be of type string, alphanumeric and lowercase with dashes only but got 111.
           - Expected \\"main\\" to be of type string but got 1333.
           - Expected \\"usage_model\\" field to be one of [\\"bundled\\",\\"unbound\\"] but got \\"INVALID\\"."
       `);
     });
 
+    it("should error on invalid `name` value with spaces", () => {
+      const expectedConfig: RawEnvironment = {
+        name: "NCC 1701 D",
```

Comment:
could you also add a test for something with an underscore to make sure that it _does_ pass?

Reply:
absolutely